tests/conftest.py

```diff
@@ -37,21 +37,41 @@ def test_dir() -> Path:
     return Path(__file__).parent
 
 
-@pytest.fixture
-def test_settings(test_dir):
-    """Return test GoogleCloudSettings object."""
-    return GoogleCloudSettings(_env_file=test_dir / "test.env")
+class TestTracker(MlflowInterface):
+    """Create sub-class of `MlflowInterface` to use in testing context.
+
+    Test class creates an in-memory sqlite db for tracking, and a temporary directory
+    for artifact storage.
+    """
+
+    def _get_tracking_password(self):
+        return "password"
 
 
 @pytest.fixture
-def test_mlflow_init_func(test_settings):
-    """Return a function that can replace ``initialize_mlflow`` with no external calls."""
+def test_tracker_factory(tmp_path):
+    def factory(experiment_name: str) -> TestTracker:
+        return TestTracker(
+            artifact_location=str(tmp_path),
+            tracking_uri="sqlite:///:memory:",
+            experiment_name=experiment_name,
+            project="",
+        )
+
+    return factory
 
-    def _test_init():
-        with unittest.mock.patch(
-            "mozilla_sec_eia.utils.cloud._access_secret_version",
-            new=lambda *args: "password",
-        ):
-            return initialize_mlflow(test_settings)
 
-    return _test_init
+@pytest.fixture
+def get_most_recent_mlflow_run_factory():
```

Comment:
non-blocking: what do you think of making `_get_run` just a normal function that can get called, instead of involving it in the fixture machinery? the fixture machinery means you don't need to import it, I guess, but the readability/complexity cost of adding this as a factory function to work around fixture mechanics seems not worth it to me.

Reply:
I think that makes sense, changed now!